src/VisualStudioExtension/QSharpVsix/QsSmartIndent.cs

```diff
@@ -125,7 +126,13 @@ private string CreateIndentation(int indent)
         /// <summary>
         /// Returns true if the character is an indentation character (a space or a tab).
         /// </summary>
-        private static bool IsIndentation(char c) => c == ' ' || c == '\t';
+        private static bool IsIndentationChar(char c) => c == ' ' || c == '\t';
+
+        /// <summary>
+        /// Returns true if the column occurs within the indentation region at the beginning of the line.
+        /// </summary>
+        private static bool IsInIndentation(ITextSnapshotLine line, int column) =>
+            line.GetText().Substring(0, column).All(IsIndentationChar);
```

Comment:
Do `ITextSnapshotLine`s have carriage returns or line feeds? If so `char.IsWhiteSpace` might be a better fit for `IsIndentationChar`

Reply:
`GetText` excludes line breaks. I didn't use `char.IsWhiteSpace` originally since I wasn't sure if I really wanted all Unicode whitespace characters to be counted as indentation... but I think it's probably fine, so I removed `IsInIndentation`.